lib/Onyx.ts

```diff
@@ -594,6 +594,42 @@ function clear(keysToPreserve: OnyxKey[] = []): Promise<void> {
         .then(() => undefined);
 }
 
+function updateSnapshots(data: OnyxUpdate[]) {
+    const snapshotCollectionKey = OnyxUtils.getSnapshotKey();
+    if (!snapshotCollectionKey) return;
+
+    const promises: Array<() => Promise<void>> = [];
+    const finalPromise = Promise.resolve();
+
+    const snapshotCollection = OnyxUtils.getCachedCollection(snapshotCollectionKey);
+
+    data.forEach(({key, value}) => {
+        // snapshots are normal keys so we want to skip update if they are written to Onyx
+        if (OnyxUtils.isCollectionMemberKey(snapshotCollectionKey, key)) {
+            return;
+        }
+
+        Object.entries(snapshotCollection).forEach(([snapshotKey, snapshotValue]) => {
+            // Snapshots may not be present in cache. We don't know how to update them so we skip.
+            if (!snapshotValue) {
+                return;
+            }
+
+            if (typeof snapshotValue !== 'object' || !('data' in snapshotValue)) return;
+
+            const snapshotData = snapshotValue.data;
+            if (!snapshotData || !snapshotData[key]) {
+                return;
+            }
+
+            const updated = _.pick(value, Object.keys(snapshotData[key]));
+            promises.push(() => merge(snapshotKey, {data: {[key]: updated}}));
```

Comment:
I'm worried that this line may impact the performance of the app, especially with a lot of updates to snapshot object